url7f15a160d2_timofei7_old_ai_assignments_tgz-pJ8-org_json_TestJ8/src/hw4/HW4Main.java

```diff
@@ -0,0 +1,318 @@
+package hw4;
+
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.Set;
+
+
+/**
+ *  main driver for hw4
+ * @author tim modified from Devin/Jiawei
+ */
+public class HW4Main
+{
+	private static final String names = "ABCDEFGHIJKLMNOPQRSTUVXYZ";
+	private static int algo;
+
+	/**
+	 * @param args
+	 */
+	public static void main(String[] args)
+	{
+		
+		Scanner input = new Scanner(System.in);
+
+		while (true)
+		{
+			System.out.print("Choose reg/mrv/lcv/print CNF (1,2,3,4) or 0 to quit: ");
+			algo = input.nextInt();
+			
+			if (algo == 0)
+			{
+				System.out.println("bailing!");
+					System.exit(1);
+			}
+			
+			ArrayList <CircuitProblem> problems = CircuitProblem.loadCircuitProblems("hw4/example.txt");
+			
+//			for(CircuitProblem cp:problems)
+//			{
+//				System.out.println(cp);
+//			}
+			
+			int i = 1;
+			for(CircuitProblem cp:problems)
+			{
+				System.out.println("Problem " + i++ + ": "+solve(cp));
+			}			
+		}
+				
+		
+	}
+	
+	/**
+	 * does the main work to solve a circuitproblem
+	 * @param cp
+	 * @return
+	 */
+	public static boolean solve(CircuitProblem cp)
+	{
+		String variables = "";
+		String values = "";
+		Solution found = new Solution(false, null);
+		
+		//build variables string
+		for (int i = 0; i < cp.size; i++)
+		{
+			if (variables=="") variables = variables + names.charAt(i);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.